### ISSUE TITLE
How to easily use the new version of VMamba?

### ISSUE THREAD

**Issue by TicTacToePP** [MzeroMiko/VMamba#22]
Hi! Thanks for your great work. I have been testing your old version (vssm.py), it seems to use the [forward_corev0](https://github.com/MzeroMiko/VMamba/blob/875c3d222c1f37c2a08fdae2759114c7c7dc58dd/classification/models/vmamba.py#L322) in the vmamba.py in the current version. So if I want to use your new version (which seems to be [this forward_corev1?](https://github.com/MzeroMiko/VMamba/blob/875c3d222c1f37c2a08fdae2759114c7c7dc58dd/classification/models/vmamba.py#L415)), Can I just simply change the [forward core here?](https://github.com/MzeroMiko/VMamba/blob/875c3d222c1f37c2a08fdae2759114c7c7dc58dd/classification/models/vmamba.py#L496-L497) It seems these 2 lines do not have (self.) before them.

**Comment by MzeroMiko** [comment]
The writing style `forward_core = self.forward_corev1` is the same as in torch.nn.Module, where you can find `__call__ = self.forward` or something like that. That means when you call `class.forward_core`, you are calling `class.forward_corev1`.

You can initialize the model with hyper-parameters `downsample_version="v1", dt_rank="auto", "ssm_ratio=2.0, mlp_ratio=0.0"` to make the new version model acts basically the same like the old one. 

If you want to infer with the model in current version, and the performance needs to be exactly the same, you would better  change `forward_core = self.forward_corev1` to  `forward_core = self.forward_corev0`. They are different only in the torch.dtype precise.

If you want to train a model yourself, and you want the parameter random initialization to be the same (with the same random seed), check `vmamba.py#check_vssm1_equals_vssm` for details. The key lies in that with the tricks used in that function, you can keep the order when `Modules` are initialized, then the random sequence will keep the same.

We hope that helps.

**Comment by TicTacToePP** [comment]
Thanks for your quick response!! Actually, I want to modify the VSSM in your old 20240128 version to use the forward_corev1. I am wondering if changing the old forward_core() in vsm.py to the new forward_corev1() in vmamba.py is enough.

**Comment by MzeroMiko** [comment]
Oh, I know what you mean now.  It should works.

**Comment by TicTacToePP** [comment]
Thanks very much! I'll try it.
